pkg/util/sysutil/io_unix.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2017 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+// +build !windows
+// +build !plan9
+
+package sysutil
+
+import (
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+const DefaultCreateMode = os.FileMode(0660)
+
+// OpenFile wraps os.OpenFile, but will always call `chmod` when creating
+// a file to ensure that it's always created with the requested permissions,
+// regardless of umask. The intention of this function is to ensure that
+// CockroachDB only creates files that are not readable by users outside of
+// the process user and group.
+func OpenFile(name string, flag int, perm os.FileMode) (*os.File, error) {
+	file, err := os.OpenFile(name, flag, perm)
+	if err != nil {
+		return nil, err
+	}
+
+	if flag&os.O_CREATE == os.O_CREATE {
+		err = file.Chmod(perm)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return file, nil
+}
+
+// Create wraps os.Create, but will use the default permissions
+func Create(name string) (*os.File, error) {
+	file, err := os.Create(name)
```

Comment:
I think it might be good to document a bit of exactly what sematics we want out of these helpers.

For example,`os.Create` will use mode `0666` to create the file.  So, if someone were to use this to create files that have predictable paths that they expect to safely get default permissions, we've potentially led them astray, because I believe they would now be in a race with a potential attacker who could open the file for reading between the Create call and the Chmod call.